fighters/rosetta/src/opff.rs

```diff
@@ -12,8 +12,80 @@ unsafe fn launch_star_cancel(boma: &mut BattleObjectModuleAccessor, status_kind:
+				VarModule::set_int(fighter.battle_object, vars::rosetta::instance::TICO_Y_DIST, (VarModule::get_int(fighter.battle_object, vars::rosetta::instance::ROSA_Y)-VarModule::get_int(fighter.battle_object, vars::rosetta::instance::TICO_Y)));
+				//Teleport!
+				if status_kind == *FIGHTER_STATUS_KIND_SPECIAL_LW && !VarModule::is_flag(fighter.battle_object, vars::rosetta::instance::IS_TICO_DEAD) && VarModule::get_int(fighter.battle_object, vars::rosetta::instance::COOLDOWN) == 0 {
+					if frame == 12.0 {
+						macros::EFFECT(fighter, Hash40::new("rosetta_escape"), Hash40::new("top"), 0, 0, -3, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, true);
+						VarModule::set_int(fighter.battle_object, vars::rosetta::status::INVIS_FRAMES, 1);
+					};
+					if frame > 16.0 && frame < 19.0 {
+						HitModule::set_whole(boma, smash::app::HitStatus(*HIT_STATUS_XLU), 0);
+						VisibilityModule::set_whole(boma, false);
+						JostleModule::set_status(boma, false);	
+						let new_x = VarModule::get_int(fighter.battle_object, vars::rosetta::instance::TICO_X) as f32;
+						let new_y = VarModule::get_int(fighter.battle_object, vars::rosetta::instance::TICO_Y) as f32;
+						let pos = smash::phx::Vector3f { x: new_x, y: new_y, z: 0.0 };
+						PostureModule::set_pos(boma, &pos);
+						PostureModule::init_pos(boma, &pos, true, true);
+						VarModule::set_int(fighter.battle_object, vars::rosetta::status::INVIS_FRAMES, 2);
+					};
+					if frame == 25.0 {
+						macros::EFFECT(fighter, Hash40::new("rosetta_escape_end"), Hash40::new("top"), 0, 0, -1.5, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, true);
+						VarModule::set_int(fighter.battle_object, vars::rosetta::status::INVIS_FRAMES, 3);
+					};
+					if frame > 25.0{
+						VisibilityModule::set_whole(boma, true);
+						JostleModule::set_status(boma, true);	
+						VarModule::set_int(fighter.battle_object, vars::rosetta::status::INVIS_FRAMES, 4);
+						HitModule::set_whole(boma, smash::app::HitStatus(*HIT_STATUS_NORMAL), 0);
+					};
+					if frame > 37.0{
+						CancelModule::enable_cancel(boma);
+					};
+				} else {
+					VarModule::set_int(fighter.battle_object, vars::rosetta::instance::ROSA_X, PostureModule::pos_x(boma) as i32);
+					VarModule::set_int(fighter.battle_object, vars::rosetta::instance::ROSA_Y, PostureModule::pos_y(boma) as i32);
+					if status_kind == *FIGHTER_STATUS_KIND_SPECIAL_LW {
+						HitModule::set_whole(boma, smash::app::HitStatus(*HIT_STATUS_NORMAL), 0);
+						JostleModule::set_status(boma, true);	
+						VisibilityModule::set_whole(boma, true);
+						if frame > 37.0{
+							CancelModule::enable_cancel(boma);
+						};
+					};
+				};
+				if VarModule::get_int(fighter.battle_object, vars::rosetta::instance::COOLDOWN) > 0 {
+					VarModule::set_int(fighter.battle_object, vars::rosetta::instance::COOLDOWN,  VarModule::get_int(fighter.battle_object, vars::rosetta::instance::COOLDOWN)-1);
```

Comment:
you can just do `VarModule::dec_int(fighter.battle_object, vars::rosetta::instance::COOLDOWN);` to decrement this var by 1.